### PR TITLE
docs: add upgrade doc for 0.0.9

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

The upcoming release `0.0.9` for Azure Key Vault Provider is incompatible with Secrets Store CSI Driver versions < `v0.0.14`.
- The driver is packaged as dependency in the provider helm charts and users running `helm upgrade` from the charts repo don't require any manual actions.
- The doc details steps users need to take for manual upgrade scenario

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Please answer the following questions with yes/no**:

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

**Special Notes for Reviewers**:

### CODE REVIEW

File: docs/how-to-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Upgrade notes
+
+This document highlights the required actions for upgrading to the latest release.
+
+## Upgrading to Key Vault provider 0.0.9+
+
+**tl;dr** - :warning: `0.0.9+` release of the Azure Key Vault provider is incompatible with the Secrets Store CSI Driver versions < `v0.0.14`.
```
Comment (ritazh) [inline docs/how-to-upgrade.md:7]: nit: should we mention driver 0.0.9+ still works with older version of the provider?

File: docs/how-to-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```bash
+➜ kubectl get ds -l app=csi-secrets-store-provider-azure -o jsonpath='{range .items[*]}{.spec.template.spec.containers[0].image}{"\n"}'
+mcr.microsoft.com/k8s/csi/secrets-store/provider-azure:0.0.9
+```
+
+### If the Secrets Store CSI Driver and Azure Key Vault Provider were installed using [deployment yamls](install-yamls.md)
+
+The driver and provider need to be updated one after the other to ensure compatible versions are being run.
+
+- Update the driver by installing the yamls from [Install the Secrets Store CSI Driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver#install-the-secrets-store-csi-driver)
```
Comment (ritazh) [inline docs/how-to-upgrade.md:49]: nit: can we make this a numbered list instead to make the order of operations a bit more obvious?

File: docs/how-to-upgrade.md (hunk 1 above)
Comment (aramase) [reply docs/how-to-upgrade.md:7]: @ritazh Do you mean driver `v0.0.14` still works with releases of provider prior to `0.0.9`?

File: docs/how-to-upgrade.md (hunk 2 above)
Comment (aramase) [reply docs/how-to-upgrade.md]: Updated!

File: docs/how-to-upgrade.md (hunk 1 above)
Comment (ritazh) [reply docs/how-to-upgrade.md:7]: Yes 🤦‍♀️

File: README.md [hunk 3]
```diff
@@ -253,6 +255,17 @@ To troubleshoot issues with the csi driver and the provider, you can look at log
   kubectl logs csi-secrets-store-secrets-store-csi-driver-7x44t secrets-store
   ```
 
+### For Azure Key Vault Provider version `0.0.9+`
+
+For `0.0.9+` the provider logs are available in the provider pods. To troubleshoot issues with the provider, you can look at logs from the provider pod running on the same node as your application pod
```
Comment (ritazh) [inline README.md:260]: 👍

**Review by ritazh — APPROVED** [review]
lgtm

**Pull request merged**